packages/web/src/components/Space/usePictureInPicture.ts

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useRef } from "react";
+
+import { RemotePeer } from "./RemotePeer";
+
+interface UsePictureInPictureOptions {
+  remotePeers: RemotePeer[];
+}
+
+const pipVideo = document.createElement("video");
+const renderer = document.createElement("canvas");
+const ctx = renderer.getContext("2d");
+
+// @ts-ignore
+window.PIP_CANVAS = renderer;
+// @ts-ignore
+window.PIP_VIDEO = pipVideo;
+
+function useInterval(callback: () => void, delay: number) {
+  const savedCallback = useRef<any>();
+
+  // Remember the latest callback.
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  // Set up the interval.
+  useEffect(() => {
+    function tick() {
+      savedCallback.current();
+    }
+    if (delay !== null) {
+      let id = setInterval(tick, delay);
+      return () => clearInterval(id);
+    }
+  }, [delay]);
+}
+
+export const usePictureInPicture = ({
+  remotePeers,
+}: UsePictureInPictureOptions) => {
+  useEffect(() => {
+    const onVisibilityChanged = () => {
+      if (document.visibilityState === "visible") {
+        // @ts-ignore
+        if (document.pictureInPictureElement) {
+          console.log(">>>>>>> Exiting PIP <<<<<<<");
+          // @ts-ignore
+          document.exitPictureInPicture();
+        }
+      } else {
+        console.log(">>>>>>> Entering PIP <<<<<<<");
+        pipVideo.play();
+        // @ts-ignore
+        pipVideo.requestPictureInPicture().catch((e) => {
+          // Do nothing
+          console.log("PIP Error:", e);
```

Comment:
Sometimes there's a weird error called "You must handle user gesture blablabla", dunno what it is yet